flink-runtime/src/main/java/org/apache/flink/runtime/rpc/RpcEndpoint.java

```diff
@@ -172,6 +173,13 @@ public void runAsync(Runnable runnable) {
 		return ((MainThreadExecutor) self).callAsync(callable, timeout);
 	}
 
+	/**
+	 * Returns the class of the self gateway type.
+	 *
+	 * @return Class of the self gateway type
+	 */
+	public abstract Class<C> getSelfGatewayType();
```

Comment:
I think one can find this out via reflection. The parameter is stored in the class signature.